homeassistant/components/becker/cover.py

```diff
@@ -0,0 +1,201 @@
+        """Init the Becker device."""
+        self._becker = becker
+        self._name = name
+        self._channel = str(channel)
+        self._template = state_template
+        self._entities = entity_ids
+        self._position = position
+        self._state = True
+
+    async def async_added_to_hass(self):
+        """Register callbacks."""
+        await super().async_added_to_hass()
+
+        old_state = await self.async_get_last_state()
+        if old_state is not None:
+            self._state = old_state.state == STATE_OPEN
+
+    @property
+    def name(self):
+        """Return the name of the device as reported by tellcore."""
+        return self._name
+
+    @property
+    def unique_id(self):
+        """Return the unique id of the device - the channel."""
+        return self._channel
+
+    @property
+    def current_cover_position(self):
+        """Return current position of cover. None is unknown, 0 is closed, 100 is fully open."""
+        return self._position
+
+    @property
+    def device_class(self):
+        """Return the class of this device, from component DEVICE_CLASSES."""
+        return DEVICE_CLASS
+
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        return COVER_FEATURES
+
+    @property
+    def is_closed(self):
+        """Return true if cover is closed, else False."""
+        return self._position == CLOSED_POSITION
+
+    async def async_open_cover(self, **kwargs):
+        """Set the cover to the open position."""
+        if self._template is None:
+            self._position = OPEN_POSITION
+        await self._becker.move_up(self._channel)
+
+    async def async_open_cover_tilt(self, **kwargs):
+        """Open the cover tilt."""
+        if self._position == CLOSED_POSITION:
+            self._position = VENTILATION_POSITION
+        elif self._position == VENTILATION_POSITION:
+            self._position = OPEN_POSITION
+        await self._becker.move_up_intermediate(self._channel)
+
+    async def async_close_cover(self, **kwargs):
+        """Set the cover to the closed position."""
+        if self._template is None:
+            self._position = CLOSED_POSITION
+        await self._becker.move_down(self._channel)
+
+    async def async_close_cover_tilt(self, **kwargs):
+        """Close the cover tilt."""
+        if self._position == OPEN_POSITION:
+            self._position = INTERMEDIATE_POSITION
+        elif self._position == INTERMEDIATE_POSITION:
+            self._position = CLOSED_POSITION
+        await self._becker.move_down_intermediate(self._channel)
+
+    async def async_stop_cover(self, **kwargs):
+        """Set the cover to the stopped position."""
+        if self._template is None:
+            self._position = 50
+        await self._becker.stop(self._channel)
+
+    async def async_update(self):
+        """Update the position of the cover."""
+        if self._template is not None:
```

Comment:
Turn this into a guard, so the rest of the function doesn't need to be indented.
```suggestion
        if self._template is None:
            return
```